### PR TITLE
Migrated projects to .NET 7

### PR DESCRIPTION
**Pull request by formlesstree4**
Plus a few other goodies~

:]

### CODE REVIEW

**Comment by ChaseDRedmon** [comment]
Thank you for the adding the additional documentation that was missing on the request types!

File: src/CatBox.NET/TemporaryFileUploadRequest.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+/// <summary>
+/// A temporary request for a collection of one or more files
+/// </summary>
+public record TemporaryFileUploadRequest
+{
+
+    /// <summary>
+    /// When the image, or images, should be expired
+    /// </summary>
+    public required ExpireAfter ExpireAfter { get; init; }
```
Comment (ChaseDRedmon) [inline src/CatBox.NET/TemporaryFileUploadRequest.cs:12]: - [ ] `Expiry` sounds better here

File: src/CatBox.NET/TemporaryStreamUploadRequest.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+/// <summary>
+/// A temporary request for an individual file upload
+/// </summary>
+public record TemporaryStreamUploadRequest
+{
+
+    /// <summary>
+    /// When the image should be expired
+    /// </summary>
+    public required ExpireAfter ExpireAfter { get; init; }
```
Comment (ChaseDRedmon) [inline src/CatBox.NET/TemporaryStreamUploadRequest.cs:12]: - [ ]  `Expiry` sounds better here

File: src/CatBox.NET/UploadHost.cs
Comment (ChaseDRedmon) [inline src/CatBox.NET/UploadHost.cs:1]: I'm not sure if we're going to end up using this or not. Might remove

File: src/CatBox.NET/ICatBox.cs [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+/// <remarks>Not currently implemented so don't use</remarks>
+public interface ICatBox
+{
+    public Task UploadFile();
+    public Task DeleteFile();
+    public Task CreateAlbum();
+    public Task EditAlbum();
+    public Task AddToAlbum();
+    public Task RemoveFromAlbum();
+    public Task DeleteAlbum();
```
Comment (ChaseDRedmon) [inline src/CatBox.NET/ICatBox.cs:16]: ~~We don't need `public` here, do we? I've never looked into or used DIMs before. What does that bring to the table here with this?~~

Edit: I guess that was apart of the original implementation for some reason.

File: src/CatBox.NET/DeleteFileRequest.cs [hunk 4]
```diff
@@ -0,0 +1,18 @@
+﻿namespace CatBox.NET;
+
+/// <summary>
+/// Wraps a request to delete files from the API
+/// </summary>
+public record DeleteFileRequest
+{
+
```
Comment (ChaseDRedmon) [inline src/CatBox.NET/DeleteFileRequest.cs:8]: - [ ] Remove trailing line

File: src/CatBox.NET/CatBoxConfig.cs [hunk 5]
```diff
@@ -8,10 +8,10 @@ public record CatBoxConfig
     /// <summary>
     /// URL for the catbox.moe domain
     /// </summary>
-    public Uri CatBoxUrl { get; set; }
+    public Uri? CatBoxUrl { get; set; }
```
Comment (ChaseDRedmon) [inline src/CatBox.NET/CatBoxConfig.cs:11]: What's the purpose of setting `CatBoxUrl` and `LitterboxUrl` to be nullable here? They technically don't have to be set if the user is using the other `Client`, but they are required for their respective clients

File: src/CatBox.NET/EditAlbumRequest.cs [hunk 6]
```diff
@@ -0,0 +1,33 @@
+﻿namespace CatBox.NET;
+
+/// <summary>
+/// Wraps a request to edit an existing album with new files, new title, new description
+/// </summary>
+public record EditAlbumRequest
+{
+
```
Comment (ChaseDRedmon) [inline src/CatBox.NET/EditAlbumRequest.cs:8]: - [ ] Remove trailing line

File: src/CatBox.NET/ExpireAfter.cs [hunk 7]
```diff
@@ -0,0 +1,28 @@
+﻿namespace CatBox.NET;
+
+/// <summary>
+/// Image expiry in litterbox.moe
+/// </summary>
+public enum ExpireAfter
+{
+
```
Comment (ChaseDRedmon) [inline src/CatBox.NET/ExpireAfter.cs:8]: - [ ] Remove trailing line

File: src/CatBox.NET/AlbumRequest.cs [hunk 8]
```diff
@@ -0,0 +1,29 @@
+﻿namespace CatBox.NET;
+
+/// <summary>
+/// Wraps a request to add files, remove files, or delete an album
+/// </summary>
+public record AlbumRequest
+{
+
```
Comment (ChaseDRedmon) [inline src/CatBox.NET/AlbumRequest.cs:8]: - [ ] Remove trailing line

File: src/CatBox.NET/CatBox.cs [hunk 9]
```diff
@@ -1,17 +1,7 @@
...
-    public Task DeleteFile();
-    public Task CreateAlbum();
-    public Task EditAlbum();
-    public Task AddToAlbum();
-    public Task RemoveFromAlbum();
-    public Task DeleteAlbum();
-}
-
-public class Catbox : ICatBox
+[Obsolete("Do not use at this time")]
```
Comment (ChaseDRedmon) [inline src/CatBox.NET/CatBox.cs:3]: Can the `bool` overload for Error be set here to set force a compiler error when attempting to use this class? Or does that have to be set on the constructor?

**Review by ChaseDRedmon — CHANGES_REQUESTED** [review]
There's a few things like code formatting and trailing lines present. I had a couple of questions on a few changes but this looks great. One more thing I want to suggest is moving all of the Request type classes into a Requests subfolder under the project. 

Thanks for organizing all of the code and writing the additional documentation that was missing on the requests types

**Comment by ChaseDRedmon** [comment]
We also need a NuGet Version bump in this PR. v0.4.0 should work

**Comment by ChaseDRedmon** [comment]
I organized the files into folders and I think this is good to merge now. LMK if you agree

File: src/CatBox.NET/CatBox.NET.csproj [hunk 10]
```diff
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
     <PropertyGroup>
-        <TargetFramework>netstandard2.1</TargetFramework>
```
[the remark this answers is not in the document]
Comment (formlesstree4) [reply src/CatBox.NET/CatBox.NET.csproj:4]: Can we multi-target? Not sure how hard that actually is to do...

File: src/CatBox.NET/CatBoxConfig.cs (hunk 5 above)
Comment (formlesstree4) [reply src/CatBox.NET/CatBoxConfig.cs:11]: It was basically to appease Visual Studio IntelliSense as I did not want to default initialize them.

**Review by ChaseDRedmon — APPROVED** [review]
Approved

File: src/CatBox.NET/CatBoxConfig.cs (hunk 5 above)
Comment (ChaseDRedmon) [reply src/CatBox.NET/CatBoxConfig.cs:11]: Yeah, that makes sense

File: src/CatBox.NET/CatBox.NET.csproj (hunk 10 above)
[the remark this answers is not in the document]
Comment (ChaseDRedmon) [reply src/CatBox.NET/CatBox.NET.csproj:4]: Yeah, I think so, PolySharp should handle the missing requirements for the library when on .NET Standard 2.1 and I added a preprocessor for the HttpContent.ReadAsStringAsync call to switch between the method that has a CancellationToken and the version of the method that doesn't in [b621886](https://github.com/ChaseDRedmon/CatBox.NET/commit/b621886ce89b1009d835fc8b1147d96e344261b4)

**Pull request merged**